src/status_im2/contexts/chat/lightbox/top_view.cljs

```diff
@@ -0,0 +1,75 @@
+(ns status-im2.contexts.chat.lightbox.top-view
+  (:require
+    [quo2.core :as quo]
+    [quo2.foundations.colors :as colors]
+    [react-native.core :as rn]
+    [react-native.orientation :as orientation]
+    [react-native.platform :as platform]
+    [react-native.reanimated :as reanimated]
+    [status-im2.contexts.chat.lightbox.common :as common]
+    [status-im2.contexts.chat.lightbox.style :as style]
+    [utils.datetime :as datetime]
+    [utils.re-frame :as rf]))
+
+(defn animate-rotation
+  [result screen-width screen-height
+   {:keys [rotate top-view-y top-view-x top-view-width top-view-bg]}]
+  (let [top-x (+ (/ common/top-view-height 2) (:top @common/insets-atom))]
+    (cond
+      (= result orientation/landscape-left)
+      (do
+        (common/set-val-timing rotate "90deg")
+        (common/set-val-timing top-view-y 60)
+        (common/set-val-timing top-view-x (- (/ screen-height 2) top-x))
+        (common/set-val-timing top-view-width screen-height)
+        (common/set-val-timing top-view-bg colors/neutral-100-opa-70))
```

Comment:
This way, one animation will be called after another and they will not be synced. A better solution can be created with the help of [derived values](https://docs.swmansion.com/react-native-reanimated/docs/api/hooks/useDerivedValue).

Reply:
@Parveshdhull Derived values can sometimes simplify animations, but I have chosen not to use derived values here for a few reasons:

1. Custom calculations would have to be written for each value. We have 4 values for the top-view and there will be more values for the bottom view. And rotation direction would need to be taken into account. Setting the value for each animation would be a lot simpler.
2. Reading the integer value out of the rotation value would complicate the equations even more. Reading it would be something like this: "(js/parseInt (subs (reanimated/get-shared-value rotate) 0 (- (count (reanimated/get-shared-value rotate)) 3)))"
3. Visually the animations look completely in sync.

Reply:
Hi, Something that looks good on the new device might look bad on older devices. It can't be considered a measure of performance.

Running several animations at the same time is inherently slow due to hardware limitations and we can't do anything about that. The best thing we can do is try to optimize implemented animations as much as possible.
Calling these animations one by one on the js thread doesn't look like a proper implementation to me, this way we lose the performance benefit provided by the reanimated library.

We can just have a single shared value (`orientation`), and all other values can be easily derived from that. All the calculation parameters like screen-height etc. are static values and can be passed while deriving value. This way while animating you only need to modify orientation shared value, others will change automatically.
(you can find an example [here](https://github.com/status-im/status-mobile/blob/2ce4a820d783427ce847a485538ffb59db7d13b0/src/js/shell_worklets.js#L166))

(This comment is a non-blocker performance suggestion if you feel this adds complexity to the code and we should skip it at this stage. Please feel free to do so.)

Reply:
Good point @Parveshdhull.

I think reimplementing with derived values is a considerable amount of work, so I will do it in another PR as a performance improvement issue.